connmgr/connmanager.go

```diff
@@ -288,7 +292,30 @@ func (cm *ConnManager) NewConnReq() {
 		cm.requests <- handleFailed{c, err}
 		return
 	}
-	c.Addr = addr
+
+	host, strPort, err := net.SplitHostPort(addr)
```

Comment:
Nice.  Yes, I completely agree about `addrmgr`.  I've mentioned a few times as well that it's probably the package that needs the most improvement/cleanup in the entire repository.

That seems like a reasonable idea to make it implement `net.Addr`.  Another approach would be to create a new type in addrmgr which embeds a `wire.NetAddress` and adds the `Network` method so it implements `net.Addr`.  That way the wire type wouldn't have to be directly modified and additional metadata could be added to the new type if desired as well.

Also, there is an open issue (#581) for converting it to make use of the database as you mentioned.